index.d.ts

```diff
@@ -0,0 +1,30 @@
+export interface Option {
+  id: number;
+  value?: string;
```

Comment:
value can be broader than a string (same line 21).

(I don't know how to asses in typescript that the value option parameter and the value attribute of the return object are of the same type)

Reply:
Good call! I've pushed an update that makes the type of `value` generic, so any kind of object can be used.

One issue is that the types now don't support omitting the `value` property. But I'm not sure if that even is officially supported behavior.